src/Sulu/Bundle/ContactBundle/Util/IdsHandlingTrait.php

```diff
@@ -0,0 +1,132 @@
+<?php
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\ContactBundle\Util;
+
+use Symfony\Component\PropertyAccess\PropertyAccess;
+use Symfony\Component\PropertyAccess\PropertyAccessor;
+
+/**
+ * Provides function to sort array by ids array.
+ */
+trait IdsHandlingTrait
```

Comment:
Instead of this service I would create two different services:

**Converter**

```php
interface IdConverter {
    // could imagine better names...
    public function convertIdsToGroupedIds($ids);
    public function convertGroupedIdsToIds($ids);
}
```

This service would be injected in all the classes currently using the `parseIds` method, which would be replaced by the `convertIdsToGroupedIds` method.

**Comparator**

```php
interface IdComparator {
    public function compare($a, $b);
}
```

Then this service could be used in the controller, manager or content type as follows:

```php
$ids = ['a1', 'c1', 'a2', 'c2'];

usort($list, function($a, $b) use($ids) {
    return $this->comparator->compare($a, $b);
});
```

This way would avoid using a trait.